### ISSUE TITLE
test.py cql-pytest test failing on makedirs

### ISSUE THREAD

**Issue by nyh** [scylladb/scylladb#12637]
In https://jenkins.scylladb.com/job/releng/job/Scylla-CI/3988/testReport/junit/(root)/non-boost%20tests/cql_pytest_cassandra_tests_validation_entities_date_type_test_dev_1/ (and we also saw the same failure in the past), we have a strange failure of the test:
```
=== TEST.PY STARTING TEST cql-pytest.cassandra_tests/validation/entities/date_type_test.1 ===
...
  File "/usr/lib/python3.11/site-packages/_pytest/junitxml.py", line 646, in pytest_sessionfinish
    os.makedirs(dirname)
  File "<frozen os>", line 225, in makedirs
FileExistsError: [Errno 17] File exists: '/jenkins/workspace/releng/Scylla-CI/scylla/testlog/x86_64/dev/xml/cql-pytest.cassandra_tests/validation/entities'
```

My guess is that we start multiple pytest invocations parallel, they each use Python's `os.makedirs()` to create the directory for the output xml file, and because we have multiple test files in the same subdirectories, we can end up with multiple attempts to makedirs the same directory in parallel, which supposedly isn't safe (see https://stackoverflow.com/questions/12468022/python-fileexists-error-when-making-directory).

I think this can be considered a pytest bug - it should use the exist_ok parameter of makedirs(). But since we can't fix that, one thing that test.py can do is to makedirs() on its own the output xml directory, and not rely on pytest doing it for us.

CC @kostja

**Comment by nyh** [comment]
Opened a pytest bug report: https://github.com/pytest-dev/pytest/issues/10694
But we need to work around this bug in our test.py until it is eventually fixed.

**Comment by kostja** [comment]
Let's make all the prefix directories once at start?

**Comment by xtrey** [comment]
The issue with race condition is fixed in pytest version 7.2.1. toolchain/dbuild uses pytest version 7.2.2. Additionally checked and was unable to reproduce.
